src/containers/consignment.rs

```diff
@@ -355,8 +356,23 @@ impl<const TRANSFER: bool> Consignment<TRANSFER> {
         if self.transfer != TRANSFER {
             status.add_warning(Warning::Custom(s!("invalid consignment type")));
         }
-        // TODO: check that interface ids match implementations
-        // TODO: check bundle ids listed in terminals are present in the consignment
+        // check ifaceid match implementation
+        for (iface, iimpl) in self.ifaces.iter() {
+            if iface.iface_id() != iimpl.iface_id {
+                status.add_warning(Warning::Custom(s!(
+                    "interface ids doesn't match implementations"
+                )));
+            }
+        }
+
+        // check bundle ids listed in terminals are present in the consignment
+        for bundle_id in self.terminals.keys() {
+            if !index.bundle_ids().any(|id| id == *bundle_id) {
+                status.add_warning(Warning::Custom(s!(
```

Comment:
```suggestion
                status.add_warning(Warning::Custom(format!(
```

Otherwise `bundle_id` will not get into the string